prime-router/src/main/kotlin/ActionLog.kt

```diff
@@ -139,7 +145,7 @@ class ActionLogger(val logs: MutableList<ActionLog> = mutableListOf()) {
      */
     fun getItemLogger(itemIndex: Int, trackingId: String? = null): ActionLogger {
         check(itemIndex > 0) { "Item index must be a positive number" }
-        return ActionLogger(this.logs, itemIndex, trackingId)
+        return ActionLogger(this.logs, itemIndex, trackingId, this.reportId)
```

Comment:
Currently, the logs added with an item logger only get the report if you call `setReportId` **after** the logs are added.  With this change that can get called at the start and then all down stream added logs will have the repot id